src/controllers/timeTrackingController.ts

```diff
@@ -0,0 +1,118 @@
+        this.timeTrackingEventService = timeTrackingEventService
+        this.getAuditLogs = getAuditLogService;
+        this.getAgents = getAgentsService;
+    }
+
+    saveNewTimeTrackingEvents = async (eventSearchStartDate?: Date) => {        
+        //get all users - to match time tracking submission with user
+        const agents = await this.getAgents(zendeskRequest);
+
+        //get newest tracking event from Mongo
+        const lastTimeTrackingEvent = await this.timeTrackingEventService.getNewestTimeTrackingEvent();  
+
+        //get the newest audit logs - with search date if it was passed to saveNewTimeTrackingEvents (so we can backfill older tickets in case of app failure)
+        const auditLogs = await this.getAuditLogs(eventSearchStartDate);
+
+        //filter audit logs to get the ones from the newest tracking event from mongo (or all events, if theres no data in mongo)
+        //we filter IF theres at least one event in db AND we dont have eventSearchStartDate (as otherwise we want to resave everything that's possible)
+        const newAuditLogs = lastTimeTrackingEvent && !eventSearchStartDate ? auditLogs.filter(log => new Date(log.created_at) > new Date(lastTimeTrackingEvent.created_at)) : auditLogs;
+
+        //for each audit log - get events that are related to time tracking
+        const timeTrackingEventsToSave: TimeTrackingEvent[] = [];
+
+        if (newAuditLogs.length > 0) {
+            newAuditLogs.forEach(log => {
+                const timeSpentEvent = log.events.find(ev => ev.field_name == TICKET_CUSTOM_FIELDS.TIME_SPENT.toString());
+                const totalTimeSpentEvent = log.events.find(ev => ev.field_name == TICKET_CUSTOM_FIELDS.TOTAL_TIME_SPENT.toString())
+
+                if (timeSpentEvent && totalTimeSpentEvent) {
+                    const timeTrackingEvent: TimeTrackingEvent = {
+                        created_at: new Date(log.created_at),
+                        zendeskEventId: timeSpentEvent.id,
+                        zendeskAuditLogId: log.id,
+                        userId: log.author_id,
+                        userName: agents.find(agent => agent.id === log.author_id).name,
+                        ticketId: log.ticket_id,
+                        timeSpent: parseInt(timeSpentEvent.value),
+                        totalTimeSpent: parseInt(totalTimeSpentEvent.value)
+                    }
+                    timeTrackingEventsToSave.push(timeTrackingEvent);
+                }
+            })
+            //save events in the db - if theres already an event with given eventID - skip!
+            try {
+                await this.timeTrackingEventService.saveTimeTrackingEvents(timeTrackingEventsToSave);
+            } catch (error) {
+                console.log(error)
+                //this doesnt work - leaving this for further investigation
```

Comment:
Is there something to be investigated here as per the comment?
@fszwedo

Reply:
I changed this, logging works ok